File: server/api/src/main/kotlin/io/patterueldev/singalong/realtime/RoomDataType.kt
package io.patterueldev.singalong.realtime

enum class RoomDataType(val value: String) {
    RESERVED_SONGS("reservedSongs"),
    CURRENT_SONG("currentSong"),
    PLAYER_LIST("playerList"),
    ASSIGNED_PLAYER_IN_ROOM("assignedPlayerInRoom"),
    PARTICIPANTS_LIST("participantsList"),
    ALL("all"),
    ;

    companion object {
        fun fromString(value: String): RoomDataType {
            return entries.first { it.value == value }
        }
    }
}

File: client/packages/singalong_api_client/lib/socket_events/room_data_type.dart
part of '../singalong_api_client.dart';

enum RoomDataType {
  reservedSongs,
  currentSong,
  playerList,
  assignedPlayerInRoom,
  participantsList,
  all,
  ;

  String get value {
    switch (this) {
      case reservedSongs:
        return 'reservedSongs';
      case currentSong:
        return 'currentSong';
      case playerList:
        return 'playerList';
      case assignedPlayerInRoom:
        return 'assignedPlayerInRoom';
      case participantsList:
        return 'participantsList';
      case all:
        return 'all';
    }
  }
}
